tests/conftest.py

```diff
@@ -109,3 +112,30 @@ def spectra_match_data_path_missing_smiles():
 @pytest.fixture(scope="session")
 def spectra_match_data_path_web_api_error():
     return str(ASSETS_DIR / "spectrum_matches_error_on_classyfire.csv")
+
+
+@pytest.fixture
+def response_100_spectra():
+    with open(ASSETS_DIR / "response_100_spectra.json", "r") as f:
+        response_json = json.load(f)
+
+    return response_json
+
+
+@pytest.fixture
+def mock_send_request(response_100_spectra) -> Mock:
+    def send_request(batch, *args):
+        batch_size = len(batch)
+        spectrum_ids = list(response_100_spectra["jsonData"].keys())[:batch_size]
+        response_data = {
+            "jsonData": {
+                id_: response_100_spectra["jsonData"][id_] for id_ in spectrum_ids
+            }
+        }
```

Comment:
this is all because you want to filter the response_100_spectra fixture to batch_size ?

Reply:
yeah. I was using a mocked definition of send_request before in one of the tests. But I had to mock other functions too because it was returning some dummy data in the wrong format. Like this I can get responses with the same size of the request, and the data is in the correct format